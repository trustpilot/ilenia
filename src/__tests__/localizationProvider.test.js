import React from 'react';
import 'raf/polyfill';
import { render } from 'react-testing-library';

import LocalizationProvider from '../LocalizationProvider';
import LocalizationContext from '../Context';

const locale = 'da-DK';

const translations = {
  myString: 'Dansk',
};

const fallbackTranslations = {
  myString: 'English (US)',
};

const Child = (props) => (
  <div>
    <span>{JSON.stringify(props.translations)}</span>
    <span>{props.locale}</span>
  </div>
);

test('Provider merges translations and fallback translations correctly', () => {
  const { container } = render(
    <LocalizationProvider
      locale={locale}
      translations={{ ...fallbackTranslations, ...translations }}
    >
      <LocalizationContext.Consumer>
        {(context) => <Child translations={context.translations} locale={context.locale} />}
      </LocalizationContext.Consumer>
    </LocalizationProvider>
  );

  expect(container.firstChild).toMatchSnapshot();
});
