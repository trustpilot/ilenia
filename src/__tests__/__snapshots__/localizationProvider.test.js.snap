// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Provider merges translations and fallback translations correctly 1`] = `
<div>
  <span>
    {"myString":"Dansk"}
  </span>
  <span>
    da-DK
  </span>
</div>
`;
